### ISSUE TITLE
MS Excel: NVDA report {color} on {backgroundColor} problem

### ISSUE THREAD

**Issue by surfer0627** [nvaccess/nvda#6029]
This case occur in NVDA2015.2 and later.
STR:
1. Open the attachment.
2. Move to cell I6.
  Press NVDA+f
  NVDA 2015.2 report: Black on Coral
  NVDA2015.1 report: black on orange
  sighted person say: black on orange
  note: Here NVDA detect-background matches with Excel Fill Color. I just curious why sighted person say "this is black on orange". Also, it seems difficult for a blind user to know that color coral is close to orange.
3. Move to cell F13.
  Press NVDA+f
  NVDA 2015.2 report: Black on Gold
  sighted person say: black on yellow
  note: In this case, Fill Color "orange" is selected.

**Comment by surfer0627** [comment]
[test  on backgroundColor.xlsx](https://github.com/nvaccess/nvda/files/295125/test.on.backgroundColor.xlsx)

**Comment by surfer0627** [comment]
[test  on backgroundColor.xlsx](https://github.com/nvaccess/nvda/files/295128/test.on.backgroundColor.xlsx)

**Comment by surfer0627** [comment]
Another case, NVDA indicate-backgroundColor does not match wiht Excel Fill Color.
Fill Color: purple
NVDA: Black on Dark Orchid
NVDA 2015.1: purple

**Comment by jcsteh** [comment]
CC @MichaelDCurran, @QChristensen.

**Comment by nvaccessAuto** [comment]
Incubated in 11382a94ef590232dbbc059657b8027daeb15942.

**Comment by surfer0627** [comment]
Test Next13360 in Excel2010:
1. Filled background color blue or light blue.
  NVDA got the same result "aqua-blue".
2. Filled background color red or dark red.
  NVDA got the same result "red".
Also, if the fill-backgroundColor name would match with the formatting info?
  E.G: fill backgroundColor purple
  formatting info: blue-purple

**Comment by michaelDCurran** [comment]
We have made two choices with NVDA's color support:

* 1. The colors that NVDA reports will only use combinations of the 
following well-known color names: red, brown, orange, yellow, green, 
aqua, blue, purple, pink, white, grey and black. However, NVDA will also 
distinguish both dark and pale versions of these colors.

* 2. Where possible, NVDA will report the closest color to what is 
actually displayed on the screen, rather than how it may be labeled in a 
specific application. For example if Excel's fill color of "orange" 
actually looks "orange-yellow" rather than orange, then the user has the 
right to know this.


I think though based on the most recent examples given by @surfer0627, 
our ratios for dark and pale still need to be tweaked a little.









On 16/06/2016 1:18 PM, surfer0627 wrote:
>
> Test Next13360 in Excel2010:
> 1. Filled background color blue or light blue.
> NVDA got the same result "aqua-blue".
> 2. Filled background color red or dark red.
> NVDA got the same result "red".
> Also, if the fill-backgroundColor name would match with the formatting 
> info?
> E.G: fill backgroundColor purple
> formatting info: blue-purple
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/nvaccess/nvda/issues/6029#issuecomment-226378417>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/ANf5ne-yuQtsYf0ij6fP8QDetf6qQgtmks5qMMCKgaJpZM4IsTFs>.
>

-- 
Michael Curran
Executive Director, NV Access Limited
Phone: +61 7 3149 3306
Website: http://www.nvaccess.org/
Twitter: @NVAccess
Facebook: http://www.facebook.com/NVAccess

**Comment by michaelDCurran** [comment]
Technical:
* NVDA classes a color as pale if it has a saturation of 50% or lower. 
We should change this to 60% or lower.
* We may want to add a second pale ratio (probably 30%) at which we 
combine the color with the word white, grey or black E.g. blue-grey.
* Colors with an HSV value (brightness) of 50% or below are classed by 
NVDA as dark. We should add an extra rule so that NVDA classes a color 
as "bright" if its above 75%. Anything between 75% and 50% are just normal.
* Bright pale should be replaced with light.



On 16/06/2016 1:18 PM, surfer0627 wrote:
>
> Test Next13360 in Excel2010:
> 1. Filled background color blue or light blue.
> NVDA got the same result "aqua-blue".
> 2. Filled background color red or dark red.
> NVDA got the same result "red".
> Also, if the fill-backgroundColor name would match with the formatting 
> info?
> E.G: fill backgroundColor purple
> formatting info: blue-purple
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub 
> <https://github.com/nvaccess/nvda/issues/6029#issuecomment-226378417>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/ANf5ne-yuQtsYf0ij6fP8QDetf6qQgtmks5qMMCKgaJpZM4IsTFs>.
>

-- 
Michael Curran
Executive Director, NV Access Limited
Phone: +61 7 3149 3306
Website: http://www.nvaccess.org/
Twitter: @NVAccess
Facebook: http://www.facebook.com/NVAccess

**Comment by nvaccessAuto** [comment]
Incubated in 6cd0ebac62233c542cd41b28c3ecb3092a89b565.

**Comment by surfer0627** [comment]
Test Next13365 in Excel2010:
1. Filled background color red.
		NVDA report: bright red
2. Filled background color dark red.
		NVDA report: red
3. Filled background color light blue.
		NVDA report: bright aqua-blue
4. Filled background color light green.
		NVDA report: pale yellow-green
Here, NVDA report two terms "bright" / "pale" for light colors.
Notes from colors.py:
> # bright color (HSV saturation 100% and value 100%)
> # dark color (HSV saturation 100% and value 44%)
> # pale color (HSV saturation 50% and value 72%)
